File: rest/src/main/java/de/philgenstock/choremaster/chore/ChoreController.java
package de.philgenstock.choremaster.chore;

import java.util.List;
import lombok.AllArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/chore")
@AllArgsConstructor
public class ChoreController {

  private final ChoreService choreService;

  @GetMapping("/household/{householdId}")
  public List<ChoreDto> getChoresFor(@PathVariable Long houseHoldId) {
    return choreService.getChoresFor(houseHoldId);
  }
}

File: choremaster-rest/rest/src/main/kotlin/de/philgenstock/choremaster/chore/ChoreController.kt
package de.philgenstock.choremaster.chore

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/chore")
class ChoreController(
    private val choreApplicationService: ChoreApplicationService,
) {
    @GetMapping("/household/{houseHoldId}")
    fun getChoresForHouseHold(
        @PathVariable("houseHoldId") houseHoldId: Long,
    ): List<ChoreDto> = choreApplicationService.getChoresForHouseHold(houseHoldId)
}
